### PR TITLE
Week3 adil

### PR DESCRIPTION
**Pull request by musafir928**

### CODE REVIEW

**Comment by ArcoMul** [comment]
Hi @musafir928 for two reasons I can highly recommend trying Netlify 😁 One it will be good practise to try it once and see how powerful it is. Second one, out of self interest, it makes it easier for me to check your homework (so that I don't have to install everything locally).

**Comment by musafir928** [comment]
https://frosty-elion-e48f60.netlify.com/

**Comment by musafir928** [comment]
Thank you for your review. I have tried netlify with this repo, but it didn't work, so, i made a new repo just with build folder,(google says it failed maybe because the folder larger than 10M ).

File: homework-submission/todo-app/src/component/week3Components/gender.jsx [hunk 1]
```diff
@@ -0,0 +1,5 @@
+import React from "react";
+
+const Gender = ({ gender }) => <li>{gender}</li>;
```
Comment (ArcoMul) [inline homework-submission/todo-app/src/component/week3Components/gender.jsx]: Add a cool icon! Maybe just use an emoji?

File: homework-submission/todo-app/src/component/week3Components/img.jsx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+import React from "react";
+const Img = ({ imgSrc, imgAlt }) => (
+  <li className="img">
+    {" "}
```
Comment (ArcoMul) [inline homework-submission/todo-app/src/component/week3Components/img.jsx]: What's this space for?

File: homework-submission/todo-app/src/component/week3Components/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
+import React, { useState, useEffect } from "react";
+import Name from "./name";
+import Img from "./img";
+import Region from "./region";
+import Gender from "./gender";
+import "core-js";
+
+function Week3Components() {
+  const [data, setData] = useState("");
```
Comment (ArcoMul) [inline homework-submission/todo-app/src/component/week3Components/index.js]: `data` is a quite generic name, try to give this a more descriptive name, of what the variable(s) actually contain.

File: homework-submission/todo-app/src/component/week3Components/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+    } catch (err) {
+      console.log(err);
+    }
+
+    return () => fetchData("https://uinames.com/api/?amount=10");
+  }, []);
+
+  document.body.onkeyup = e => {
+    if (e.keyCode === 32 && id <= 8) {
+      setId(id + 1);
```
Comment (ArcoMul) [inline homework-submission/todo-app/src/component/week3Components/index.js]: Not so 'random' in your case 😉

File: homework-submission/todo-app/src/component/week3Components/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  return (
+    <div className="week3">
+      {id === -1 ? (
+        <h1 className="press-space">Please press space to get new data</h1>
+      ) : (
+        <>
+          <h1>week3-homework</h1>
+          <ul>
+            <Img
+              imgSrc={`https://uinames.com/api/photos/${data[id].gender}/${id +
```
Comment (ArcoMul) [inline homework-submission/todo-app/src/component/week3Components/index.js]: If you read the api documentation you can find how to get an image url in the returned json data. A bit neater than generating the url yourself. Please give it a try :)

File: homework-submission/todo-app/src/component/week3Components/img.jsx (hunk 2 above)
Comment (musafir928) [reply homework-submission/todo-app/src/component/week3Components/img.jsx]: I don't know, when I saved the code, it was added by formatter.

File: homework-submission/todo-app/src/component/week3Components/gender.jsx (hunk 1 above)
Comment (musafir928) [reply homework-submission/todo-app/src/component/week3Components/gender.jsx]: I have fixed it in week4 homework.

File: homework-submission/todo-app/src/component/week3Components/img.jsx (hunk 2 above)
Comment (musafir928) [reply homework-submission/todo-app/src/component/week3Components/img.jsx]: But, I deleted it.

File: homework-submission/todo-app/src/component/week3Components/index.js (hunk 3 above)
Comment (musafir928) [reply homework-submission/todo-app/src/component/week3Components/index.js]: Changed to user.

File: homework-submission/todo-app/src/component/week3Components/index.js (hunk 4 above)
Comment (musafir928) [reply homework-submission/todo-app/src/component/week3Components/index.js]: HeHe, it was 'random' in a word 'ordered random'.hhh... but fixed in week4.

File: homework-submission/todo-app/src/component/week3Components/index.js (hunk 5 above)
Comment (musafir928) [reply homework-submission/todo-app/src/component/week3Components/index.js]: fixed with ext.

**Comment by ArcoMul** [comment]
Hi @musafir928, could you get your fixes into this branch/pull request? It gets a bit hard for me if I have to check multiple pull requests to find the correct version of your homework.

**Comment by musafir928** [comment]
> Hi @musafir928, could you get your fixes into this branch/pull request? It gets a bit hard for me if I have to check multiple pull requests to find the correct version of your homework.

Ok, I will push it now.

**Comment by ArcoMul** [comment]
Thanks @musafir928! Nice improvements! :-)

One little informative thing: `Access-Control-Allow-Origin` is a Response Header, meaning that the server will send it as response to the browser. Sending it as a Request Header to the server doesn't do much. Read more here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin

**Comment by musafir928** [comment]
> Thanks @musafir928! Nice improvements! :-)
> 
> One little informative thing: `Access-Control-Allow-Origin` is a Response Header, meaning that the server will send it as response to the browser. Sending it as a Request Header to the server doesn't do much. Read more here: https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin

OOOOO..!  Thank you, I was confused, now it's clear! res and req, I will change it.

**Pull request closed without merging**